### PR TITLE
Fix emacs completion returning bogus candidates.

### PR DESCRIPTION
**Pull request by chripell**
In standalone mode, utop returns all the possible completions when no
one matches what the user has typed. This is surprising in emacs
because of the different UI/UX. For this reason, we keep a list of
prefixes for which completion has started and return only the
candidates which match the prefixes.

Fixes: #343

### CODE REVIEW

**Comment by pmetzger** [comment]
@rgrinberg I presume we should merge this?

File: src/top/utop.el [hunk 1]
```diff
@@ -645,7 +648,10 @@ it is started."
        (setq utop-completion nil))
       ;; A new possible completion
       ("completion"
-       (push argument utop-completion))
+       (catch 'done
+	 (dolist (prefix utop-completion-prefixes)
+	   (if (string-prefix-p prefix argument)
+	       (progn (push argument utop-completion) (throw 'done t)) nil))))
```
Comment (rgrinberg) [inline src/top/utop.el:654]: Why do you need the `nil` at the end?

File: src/top/utop.el [hunk 2]
```diff
@@ -645,7 +648,10 @@ it is started."
        (setq utop-completion nil))
       ;; A new possible completion
       ("completion"
-       (push argument utop-completion))
+       (catch 'done
+	 (dolist (prefix utop-completion-prefixes)
```
Comment (rgrinberg) [inline src/top/utop.el:652]: The indentation seems off. `(dolist` should be indented because it's inside `catch`

File: src/top/utop.el (hunk 1 above)
Comment (chripell) [reply src/top/utop.el:654]: Sorry, leftover from a previous version which had also an else branche. Fixed.

File: src/top/utop.el (hunk 2 above)
Comment (chripell) [reply src/top/utop.el:652]: Fixed.

**Comment by chripell** [comment]
PTAL

**Comment by pmetzger** [comment]
@rgrinberg Let me know if this is now mergeable.

**Comment by pmetzger** [comment]
@rgrinberg ?

**Comment by rgrinberg** [comment]
Merged with a minor modifications. Thanks.

**Comment by chripell** [comment]
Thanks!

**Pull request merged**